rust/blockstore/src/key.rs

```diff
@@ -116,11 +124,7 @@ impl Eq for CompositeKey {}
 
 impl PartialOrd for CompositeKey {
     fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
-        if self.prefix == other.prefix {
-            self.key.partial_cmp(&other.key)
-        } else {
-            self.prefix.partial_cmp(&other.prefix)
-        }
+        Some(self.cmp(other))
```

Comment:
Little bit confused by this change, How does this work, will it apply the same comparison per field of the CompositeKey?

Reply:
That's what I reasoned to being safe.  Specifically it implements partial_cmp the same as cmp.